### DISCUSSION TITLE
Dropdowns in Editiermaske

### DISCUSSION THREAD

**Discussion by thomaskofler** [e-netze/webgis-community#130] [General]
Es soll eine Suchfunktion innerhalb von Dropdowns in der Editiermaske geben, dabei soll bei der Eingabe eines Buchstabes automatisch gefiltert werden (wie bei der Schnellsuche/Detailsuche).

**Comment by gis-eni** [comment]
Um wie viele Einträge handelt es sich hier? Warum nicht gleich ein Autocomplete verwenden?

**Reply by thomaskofler** [reply]
es wären ca. 50 Einträge; wie stellt man das Autocomplete ein?

**Reply by gis-eni** [reply]
Du stellst beim Editierthema als Eingabe Typ "autocomplete" ein:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/3afc575e-8d3c-48d2-8a4f-3ddf4d5e091a)

Ansonsten gehts du gleich wie bei einer Auswahlliste. Nur musst du der WHERE Clause den Platzhalter ``{0}`` einbauen. Der steht dann für den Wert, den der Anwender gerade eingibt, zB

``MyField like '{0}%'``

**Comment by thomaskofler** [comment]
ok danke, das probiere ich aus

**Reply by thomaskofler** [reply]
Funktioniert bei mir leider nicht, Konfiguration schaut so aus:
![grafik](https://github.com/gis-eni/webgis-forum/assets/119412172/e9ef5fca-6b1f-4487-b15f-93a0214fcb89)

**Reply by gis-eni** [reply]
Ok, du verwendest im Hintergrund eine Datalinq Abfrage. Da ist jetzt natürlich die Frage, wie deine Datalinq Abfrage abgebaut ist:

* Leitet DataLinq die Abfrage auf eine Datenbank weiter?
* Funktioniert die DataLinq Abfrage wenn du: https://www.........@-kategorie?value=.... aufrufst? Probier einmal nur einen Buchstaben zu übergeben? Kommt dann Ergebnis?
* Wie aktuell ist deine WebGIS Version?

**Reply by thomaskofler** [reply]
Nein, es wurde einfach im Datalinq eine Liste (plaintext) erstellt; beim Aufruf value=... kommt kein Ergebnis. Wir haben momentan eine ältere Version installiert (5.23.901), am Donnerstag wird die aktuelle Version installiert.

**Reply by gis-eni** [reply]
In PlainText Listen ist keine Suche implementiert. Dort kann man einfach nur alle Werte abholen. Diese Methode würd für "kleine" Auswahllisten konzipiert...

**Reply by gis-eni** [reply]
Hab das noch einmal evaluiert: 
die Einstellungen in CMS müssen so lauten:

<img width="469" alt="image" src="https://github.com/gis-eni/webgis-forum/assets/62870568/78599906-b702-477f-b555-e506c01f6959">

Allerdings: Autocomplete über eine DataLinq Abfrage funktioniert erst ab WebGIS 6. Dort kommt auch die Möglichkeit, dass man PlainText abfragen filtern kann.
